build: update cross-repo angular dependencies

alan-agius4 opened the pull request:
See associated pull request for more information.

Closes #59956 as a pr takeover, `@angular/build-tooling` has been rolled back as due to missing `@aspect_rules_js`.

alan-agius4 commented on package.json:
```diff
@@ -159,10 +159,10 @@
     "@actions/github": "^6.0.0",
-    "@angular-devkit/architect-cli": "0.1902.0",
+    "@angular-devkit/architect-cli": "0.2000.0-next.1",
     "@angular/build-tooling": "https://github.com/angular/dev-infra-private-build-tooling-builds.git#ce04ec6cf7604014191821a637e60964a1a3bb4a",
```
@devversion, should we disable renovate from update `@angular/build-tooling` or start add `@aspect_rules_js`?

devversion replied:
Yeah, let's ignore it for now.

alan-agius4 committed:
build: update cross-repo angular dependencies

See associated pull request for more information.

Closes #59956 as a pr takeover, `@angular/build-tooling` has been rolled back as due to missing `@aspect_rules_js`.

alan-agius4 commented on devtools/tools/esbuild/index.bzl:
```diff
@@ -56,6 +56,7 @@ def _create_bundle_targets(pkg, entry_point, module_name):
         output = "%s/index.mjs" % target_name_base,
         platform = pkg.platform,
         entry_point = "@npm//:node_modules/@angular/%s/%s" % (pkg.name, fesm_bundle_path),
+        deps = ["@npm//@angular/%s" % pkg.name],
```
This is needed as now the FESMs have shared chunk files.

alan-agius4 commented:
**caretake note:** the lint failures are the usual license issues.

alan-agius4 committed:
ci: disable Renovate updates for `@angular/build-tooling`

The newer version of this package requires Aspect setup which is not configured in this repo.

pkozlowski-opensource commented:
This PR was merged into the repository by commit 80895ea966ac512aec7c588d2a469a5f7b8c1c9a.

The changes were merged into the following branches: main
